### ISSUE TITLE
Unable to nest @Embedded Objects on MongoDB

### ISSUE THREAD

**Issue by monesidn** [Impetus/kundera#863]
When working with MongoDB we can not store a nested `@Embedded` object inside another `@Embedded` one.
For example, given:

``` java
@Entity
@SuppressWarnings("serial")
public class RootClass1 implements Serializable{

    @Id
    @GeneratedValue
    private ObjectId objectId;

    @Embedded
    private ParentClass1 parent;
```

``` java
@Embeddable
@SuppressWarnings("serial")
public class ParentClass1 implements Serializable {

    private String payload;

    @Embedded
    private ChildClass1 child;
```

``` java
@Embeddable
@SuppressWarnings("serial")
public class ChildClass1 implements Serializable{

    private String payload;
```

Kundera mongodb stores `RootClass1` and `ParentClass1` ignoring the `@Embedded ChildClass1` field.
An example `ParentClass1` document:
`{ "_id" : { "$oid" : "573b2dfe971871e5231fd0cc"} , "parent" : { "payload" : "Parent"}}`

I wrote two test cases that can be found at: 
https://github.com/monesidn/kundera-mongodb-test-embeddable

**Comment by devender-yadav** [comment]
Hi @monesidn,

Another user reported this issue on [gitter chat](https://gitter.im/impetus-opensource/Kundera?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge).

it's fixed has been added into the trunk and will be available in next release. Meanwhile, you can build your code from trunk and let me know if you still face any issue.

You can refer [PersonnelEmbeddedTest](https://github.com/impetus-opensource/Kundera/blob/052774f76fb14b564ce4cb729db93b93c889341f/src/kundera-mongo/src/test/java/com/impetus/client/crud/PersonnelEmbeddedTest.java) for details.

-Dev

**Comment by monesidn** [comment]
Thank you @devender-yadav. I've successfully run the test against the 3.5-SNAPSHOT (the current trunk). 
I just need another information: when the 3.5 release is scheduled? I haven't been able to find this information anywhere.

**Comment by devender-yadav** [comment]
Hi @monesidn 

Kundera 3.5 release is planned for 16 June 2016. Check details about milestone 3.5 [here](https://github.com/impetus-opensource/Kundera/milestones) .

-Dev

**Comment by devender-yadav** [comment]
Hi @monesidn,

Fixed in Kundera 3.5

-Dev
